pwiz_tools/Skyline/EditUI/AssociateProteinsDlg.cs

```diff
@@ -178,36 +190,84 @@ public void UseFastaFile(string file)
         }
 
         // creates dictionary of a protein to a list of peptides that match
-        private void FindProteinMatchesWithFasta(Stream fastaFile)
+        private List<KeyValuePair<FastaSequence, List<IdentityPath>>> FindProteinMatchesWithFasta(Stream fastaFile)
         {
-            var proteinAssociations = new List<KeyValuePair<FastaSequence, List<PeptideDocNode>>>();
             var peptidesForMatching = ListPeptidesForMatching();
-            using (var reader = new StreamReader(fastaFile))
+            if (peptidesForMatching.Count == 0)
+            {
+                return new List<KeyValuePair<FastaSequence, List<IdentityPath>>>();
+            }
+
+            int maxLength = peptidesForMatching.Max(peptide => GetPeptideSequence(peptide).Length);
+            long streamLength = fastaFile.Length;
+            var proteinAssociations = new List<Tuple<int, FastaSequence, List<IdentityPath>>>();

```

Comment:
I really hate Tuple. Can you just define a struct with named properties for this? It makes the code so much easier to read when you return to it later or for the first read for someone not as immersed in your intent.